tests/integration.rs

```diff
@@ -53,31 +53,31 @@ fn main() {
             last_log_location: Mutex::new(None),
         });
         let a = me.clone();
-        set_boxed_logger(Box::new(Logger(me))).unwrap();
+        let logger = Logger(me);
```

Comment:
What happens if we call a log macro with `logger` param here? The macro would expand a `let logger = $logger;`, which seems to move rather than borrow the logger variable?

Reply:
IIUC it would be `debug!(logger: &logger, ...)` then. That is, use ref from the user size.

One of the rationales is @EFanZh's comment about "binary size", see:

* https://github.com/rust-lang/log/pull/664#discussion_r1972703604
* https://github.com/rust-lang/log/pull/664#discussion_r1973798743

Reply:
I see. If explicitly passing ref from the user side is intentional, then I'm fine with that.

Reply:
@KodrAus: How about keeping `__private_api::log` taking logger argument by value, but the logger macros passes references for custom loggers, and passes global logger by value? In this way, the global logger can still utilize the ZST optimization, although not the custom ones.

Reply:
@EFanZh That sounds good to me :+1: We're currently halfway there; the `__private_api::log` function takes the logger by value, but the way we expand `log!` macros always takes by reference. I'll follow up with that change to make an unspecified logger accept `GlobalLogger` by value. That's also a good way for me to get fully across the new code.